dxp/layout_rendering.md

```diff
@@ -0,0 +1,173 @@
+3. `/c/portal/layout?p_l_id=19&p_v_l_s_g_id=0`: this is internal too. The server maps the `/web/guest` request (which is a virtual path) to the configured layout, identified by a `p_l_id`. This is the request that truly generates the HTML content the user sees.
+4. `/o/js_loader_config?t=1600876854212`: this is a external request triggered by the HTML once the browser loads it. It returns the configuration of the AMD loader.
+5. `/o/classic-theme/js/main.js?browserId=firefox&minifierType=js&languageId=en_US&b=7305&t=1600791970000`: external request downloading the theme's JavaScript code.
+6. `/combo?browserId=firefox&minifierType=js&languageId=en_US&b=7305&t=1600873127973&/o/frontend-js-aui-web/aui/aui/aui.js&...&/o/frontend-js-web/liferay/workflow.js`: external request downloading DXP's core JavaScript code.
+7. `/o/js_bundle_config?t=1600876893358`: external request downloading the configuration of the legacy AUI loader.
+8. `/o/js_resolve_modules?...`: a request to resolve AMD modules. Several more of this follow as well as requests to the `combo` servlet to download the resolved modules.
+9. `/image/company_logo`: request for web site logo.
+10. `/o/classic-theme/images/clay/icons.svg`: request for Clay icon library.
+11. `/c/portal/login?p_l_id=19&windowState=exclusive`: internal request for the login portlet.
+12. `/web/guest/home-widget-?p_p_id=com_liferay_login_web_portlet_LoginPortlet&p_p_lifecycle=0&p_p_state=exclusive&p_p_mode=view&saveLastPath=false&_com_liferay_login_web_portlet_LoginPortlet_mvcRenderCommandName=%2Flogin%2Flogin`: the real path servicing the login's portlet response.
+
+In this document we are mainly interested in the first 3 which are the responsibles for rendering the HTML code you see in the browser. Specifically the third one, as the other two are simply forwards.
+
+# How layout requests are processed
+
+There are several pieces involved in rendering a layout. We will see the details of each one in the following sections.
+
+## `MainServlet`
+
+Processes any request to `/c/...` URLs. It reads the `/WEB-INF/struts-config.xml` file to configure itself with `Action`s, `ActionForward`s, ...
+
+## `PortalRequestProcessor`
+
+Processes the following paths (see the [code]()):
+
+```java
+	_publicPaths.add(_PATH_C);
+	_publicPaths.add(_PATH_PORTAL_API_JSONWS);
+	_publicPaths.add(_PATH_PORTAL_FLASH);
+	_publicPaths.add(_PATH_PORTAL_J_LOGIN);
+	_publicPaths.add(_PATH_PORTAL_LAYOUT);
+	_publicPaths.add(_PATH_PORTAL_LICENSE);
+	_publicPaths.add(_PATH_PORTAL_LOGIN);
+	_publicPaths.add(_PATH_PORTAL_RENDER_PORTLET);
+	_publicPaths.add(_PATH_PORTAL_TCK);
+	_publicPaths.add(_PATH_PORTAL_UPDATE_LANGUAGE);
+	_publicPaths.add(_PATH_PORTAL_UPDATE_PASSWORD);
+	_publicPaths.add(_PATH_PORTAL_VERIFY_EMAIL_ADDRESS);
+	_publicPaths.add(PropsValues.AUTH_LOGIN_DISABLED_PATH);
+```
+
+`MainServlet` delegates to `PortalRequestProcessor.process()` the handling of every `GET` and `POST` request.
```

Comment:
> `MainServlet` delegates to `PortalRequestProcessor.process()` the handling of every `GET` and `POST` request.

That sounds a bit like Yoda, so I'd reorder:

> `MainServlet` delegates the handling of every `GET` and `POST` request to `PortalRequestProcessor.process()`.

Reply:
![yoda](https://media.giphy.com/media/6fScAIQR0P0xW/giphy.gif)